fbgemm_gpu/CMakeLists.txt

```diff
@@ -330,7 +435,9 @@ if(NVML_LIB_PATH)
   target_link_libraries(fbgemm_gpu_py ${NVML_LIB_PATH})
 endif()
 target_include_directories(fbgemm_gpu_py PRIVATE ${TORCH_INCLUDE_DIRS})
-set_property(TARGET fbgemm_gpu_py PROPERTY CXX_STANDARD 17)
+if(USE_CUDA)
+  set_property(TARGET fbgemm_gpu_py PROPERTY CXX_STANDARD 17)
+endif()
```

Comment:
Is this a CUDA-specific property?

Reply:
ROCm c++ standard is set on https://github.com/ROCmSoftwarePlatform/FBGEMM/blob/eabd0a8723efff806eb3c4e8947ee754f56dfdba/fbgemm_gpu/cmake/Hip.cmake#L136

Reply:
Can't ROCm also use C++17 (as CUDA does)? Or is there any reason to stick to C++14? I am not sure if FBGEMM stay compilable with C++14 in the future.

Reply:
> Can't ROCm also use C++17 (as CUDA does)? Or is there any reason to stick to C++14? I am not sure if FBGEMM stay compilable with C++14 in the future.

We are using c++ 17, please see the link above.

Reply:
If ROCm is also using C++17, do you need to add this `if (USE_CUDA)` branch?

Reply:
'hip_add_library' has a different way to set the compilation flags with `HIPCC_OPTIONS` (https://github.com/ROCmSoftwarePlatform/FBGEMM/blob/99a70e143e7ee21565d17cee2c66930ba743253e/fbgemm_gpu/CMakeLists.txt#L419). We can remove this `if (USE_CUDA)` condition but leave it as an ineffective statement. Would you like me to remove it? Thank you @shintaro-iwasaki

Reply:
We should minimize the gap if it does not cause a problem (e.g., error or warning). Can we remove this condition if ROCm will also use C++17?

Reply:
OK. will do.